src/grasp_generator.cpp

```diff
@@ -1046,50 +1050,107 @@ bool GraspGenerator::generateSuctionGrasps(const Eigen::Affine3d& cuboid_top_pos
     visual_tools_->trigger();
     visual_tools_->prompt("Added center grasp, continue?");
   }
-
+  grasp_poses.push_back(grasp_pose);
   addGrasp(grasp_pose, grasp_data, grasp_candidates, cuboid_top_pose, 0);
 
   // We define min, max and inc for each for loop here for readability
-  double x_min = -depth / 2.0 + grasp_data->active_suction_range_x_ / 2.0;
-  double x_max = depth / 2.0 - grasp_data->active_suction_range_x_ / 2.0;
-  double x_inc = grasp_data->grasp_resolution_;
 
-  double y_min = -width / 2.0 + grasp_data->active_suction_range_y_ / 2.0;
-  double y_max = width / 2.0 - grasp_data->active_suction_range_y_ / 2.0;
-  double y_inc = grasp_data->grasp_resolution_;
+  // if X range is less than y range then we use x range for the xy range
+  double xy_min, xy_max, xy_inc;
```

Comment:
```suggestion
  double xy_min, xy_max, xy_increment;
```

let's not over abbreviate, that makes for very difficult code to read